tests/TestCase/Collection/CollectionTest.php

```diff
@@ -85,15 +85,23 @@ public function testAvg($items)
         $collection = new Collection($items);
         $this->assertEquals(2, $collection->avg());
 
-        $collection = new Collection([]);
-        $this->assertNull($collection->avg());
-
         $items = [['foo' => 1], ['foo' => 2], ['foo' => 3]];
         $collection = new Collection($items);
         $this->assertEquals(2, $collection->avg('foo'));
     }
 
     /**
+     * Tests the avg method when on an empty collection
+     *
+     * @return void
+     */
+    public function testAvgWithEmptyCollection()
+    {
+        $collection = new Collection([]);
+        $this->assertNull($collection->avg());
```

Comment:
According to https://math.stackexchange.com/questions/909395/what-is-the-average-of-no-numbers the average without numbers doesn't exist.

But I can add it.